bench/workgen/workgen.cpp

```diff
@@ -476,40 +489,50 @@ WorkloadRunner::start_tables_create(WT_CONNECTION *conn)
     }
 
     ContextInternal *icontext = _workload->_context->_internal;
-    bool manage_db_size =
-      _workload->options.create_target > 0 && _workload->options.create_trigger > 0;
-    uint32_t db_size = 0;
+    int create_target = _workload->options.create_target;
+    int create_trigger = _workload->options.create_trigger;
+    int max_files = _workload->options.max_num_files;
+    bool manage_db_size = create_target > 0 && create_trigger > 0;
     bool creating = true;
 
     if (manage_db_size) {
-        db_size = get_dir_size_mb(_wt_home);
-        // Initially we start creating tables if the database size is less than the create target.
-        creating = db_size < _workload->options.create_target;
+        uint32_t db_size = get_dir_size_mb(_wt_home);
+        uint32_t num_files = get_dir_num_files(_wt_home);
+        /*
+         * Initially we start creating tables if:
+         *  - the database size is less than the create target and
+         *  - then number of files is below the limit.
```

Comment:
should this be "the" not "then"

Reply:
Thank you! Fixed in 75c489a864a596812c886b543791ff2ed68ce89d.